dotnet/src/Connectors/Connectors.HuggingFace/TextEmbedding/HuggingFaceTextEmbeddingGenerationService.cs

```diff
@@ -16,137 +16,46 @@ namespace Microsoft.SemanticKernel.Connectors.HuggingFace;
-    /// <param name="model">Model to use for service API call.</param>
-    /// <param name="endpoint">Endpoint for service API call.</param>
-    public HuggingFaceTextEmbeddingGenerationService(string model, string endpoint)
-    {
-        Verify.NotNullOrWhiteSpace(model);
-        Verify.NotNullOrWhiteSpace(endpoint);
-
-        this._model = model;
-        this._endpoint = endpoint;
-        this._attributes.Add(AIServiceExtensions.ModelIdKey, this._model);
-        this._attributes.Add(AIServiceExtensions.EndpointKey, this._endpoint);
-        this._httpClient = HttpClientProvider.GetHttpClient();
-    }
+    /// <inheritdoc />
+    public IReadOnlyDictionary<string, object?> Attributes => this.AttributesInternal;
 
     /// <summary>
-    /// Initializes a new instance of the <see cref="HuggingFaceTextEmbeddingGenerationService"/> class.
+    /// Initializes a new instance of the <see cref="HuggingFaceTextGenerationService"/> class.
     /// </summary>
-    /// <param name="model">Model to use for service API call.</param>
-    /// <param name="httpClient">The HttpClient used for making HTTP requests.</param>
-    /// <param name="endpoint">Endpoint for service API call. If not specified, the base address of the HTTP client is used.</param>
-    public HuggingFaceTextEmbeddingGenerationService(string model, HttpClient httpClient, string? endpoint = null)
+    /// <param name="model">The HuggingFace model for the text generation service.</param>
+    /// <param name="endPoint">The endpoint uri including the port where HuggingFace server is hosted</param>
+    /// <param name="apiKey">Optional API key for accessing the HuggingFace service.</param>
+    /// <param name="httpClient">Optional HTTP client to be used for communication with the HuggingFace API.</param>
+    /// <param name="loggerFactory">Optional logger factory to be used for logging.</param>
+    public HuggingFaceTextEmbeddingGenerationService(
+        string model,
+        Uri? endPoint = null,
+        string? apiKey = null,
+        HttpClient? httpClient = null,
+        ILoggerFactory? loggerFactory = null)
     {
         Verify.NotNullOrWhiteSpace(model);
-        Verify.NotNull(httpClient);
-        if (httpClient.BaseAddress == null && string.IsNullOrEmpty(endpoint))
-        {
-            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
-        }
 
-        this._model = model;
-        this._endpoint = endpoint;
-        this._httpClient = httpClient;
-        this._attributes.Add(AIServiceExtensions.ModelIdKey, model);
-        this._attributes.Add(AIServiceExtensions.EndpointKey, endpoint ?? httpClient.BaseAddress!.ToString());
+        this.Client = new HuggingFaceClient(
+            modelId: model,
+            apiKey: apiKey,
+            httpRequestFactory: new HuggingFaceHttpRequestFactory(),
+#pragma warning disable CA2000
```

Comment:
Useful to include the description of the warning

`#pragma warning disable CA2000 // Dispose objects before losing scope`